neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import Dict
+import numpy as np
+import math
+
+import torch
+import torch.nn as nn
+
+from neuralhydrology.modelzoo.head import get_head
+from neuralhydrology.modelzoo.basemodel import BaseModel
+
+LOGGER = logging.getLogger(__name__)
+
+class Transformer(BaseModel):
+    """Transformer model class, which relies on PyTorch's TransformerEncoder class.
+
+    This class implements the encoder of a transformer network with a regression or probabilistic head. 
```

Comment:
We still don't have probabilistic heads, so I would not mention that here.
```suggestion
    This class implements the encoder of a transformer network which can be used for regression.
```